blockchain/validate.go

```diff
@@ -1243,27 +1247,44 @@ func (b *BlockChain) checkConnectBlock(node *blockNode, block *btcutil.Block, vi
 	return nil
 }
 
-// CheckConnectBlock performs several checks to confirm connecting the passed
-// block to the main chain does not violate any rules.  An example of some of
-// the checks performed are ensuring connecting the block would not cause any
-// duplicate transaction hashes for old transactions that aren't already fully
```

Comment:
Please leave the meat of the description here.  Unexported functions don't show up on godoc, so when looking at https://godoc.org/github.com/btcsuite/btcd/blockchain, the information will not be visible.  That is typically why the unexported comments refer to the exported comments for more details.

Reply:
This doesn't seem like a good function to leave a docstring enumerating all block validation rules (since now the function performs more checks than just these). I feel like it is sufficient to say "CheckConnectBlockTemplate fully validates that connecting the passed block to the main chain does not violate any consensus rules, aside from the proof of work requirement", which it does.

But if you feel like  it belongs on here, I'm willing to move it back.

Reply:
Fair enough.  I'm not overly concerned about it here.  I just noticed that the bulk of the comment was moved to the unexported function.  I agree that calling out it validates per the consensus rules is probably good enough.